src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -708,6 +711,43 @@ class WorkspaceService(userInfo: UserInfo, dataSource: DataSource, workspaceDAO:
       }
     }
 
+  /**
+   * Munges together the output of Cromwell's /outputs and /logs endpoints, grouping them by task name */
+  private def mergeWorkflowOutputs(execOuts: ExecutionServiceOutputs, execLogs: ExecutionServiceLogs, workflowId: String): PerRequestMessage = {
+    val outs = execOuts.outputs
+    val logs = execLogs.logs
+
+    val outsByTask = outs groupBy { case (k,_) => k.split('.').dropRight(1).mkString(".") }
+
+    val taskMap = (outsByTask.keySet ++ logs.keySet).map( key => key -> TaskOutput( logs.get(key), outsByTask.get(key)) ).toMap
+    RequestComplete(StatusCodes.OK, WorkflowOutputs(workflowId, taskMap))
+  }
+
+  /**
+   * Get the list of outputs for a given workflow in this submission */
+  def workflowOutputs(workspaceName: WorkspaceName, submissionId: String, workflowId: String) = {
+    dataSource inTransaction { txn =>
+      withWorkspaceContextAndPermissions(workspaceName, GCSAccessLevel.Read, txn) { workspaceContext =>
+        withSubmission(workspaceContext, submissionId, txn) { submission =>
+          withWorkflow(submission, workflowId) { workflow =>
+
+            val mergedOutputs:Try[PerRequestMessage] = for {
+              outsRq <- Try(executionServiceDAO.outputs(workflowId, userInfo.authCookie))
+              logsRq <- Try(executionServiceDAO.logs(workflowId, userInfo.authCookie))
+            } yield mergeWorkflowOutputs(outsRq, logsRq, workflowId)
```

Comment:
This was a suggestion from @geoffjentry so I defer to him on semantic questions. `mergeWorkflowOutputs` is intended *not* to run if the Try fails.